src/Microsoft.Data/PrimitiveDataFrameColumnContainer.cs

```diff
@@ -0,0 +1,158 @@
+        public PrimitiveDataFrameColumnContainer(T[] values)
+        {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            long ii = 0;
+            long length = values.LongLength;
+            DataFrameBuffer<T> curBuffer;
+            if (Buffers.Count == 0)
+            {
+                curBuffer = new DataFrameBuffer<T>();
+                Buffers.Add(curBuffer);
+            }
+            else
+            {
+                curBuffer = Buffers[Buffers.Count - 1];
+            }
+            for (; ii < length; ii++)
+            {
+                if (curBuffer.Length == curBuffer.MaxCapacity)
+                {
+                    curBuffer = new DataFrameBuffer<T>();
+                    Buffers.Add(curBuffer);
+                }
+                curBuffer.Append(values[ii]);
+                Length++;
+            }
+        }
+        public PrimitiveDataFrameColumnContainer(IEnumerable<T> values)
+        {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            if (Buffers.Count == 0)
+            {
+                Buffers.Add(new DataFrameBuffer<T>());
+            }
+            var curBuffer = Buffers[Buffers.Count - 1];
+            foreach (var value in values)
+            {
+                if (curBuffer.Length == curBuffer.MaxCapacity)
+                {
+                    curBuffer = new DataFrameBuffer<T>();
+                    Buffers.Add(curBuffer);
+                }
+                curBuffer.Append(value);
+                Length++;
+            }
+        }
+        public PrimitiveDataFrameColumnContainer() { }
+        public long Length;
+        //TODO:
+        public long NullCount => throw new NotImplementedException();
+        private int GetArrayContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentException($"Index {rowIndex} cannot be greater than the Column's Length {Length}");
+            }
+            int curArrayIndex = 0;
+            int numBuffers = Buffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > Buffers[curArrayIndex].Length)
+            {
+                rowIndex -= Buffers[curArrayIndex].Length;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        public IList<T> this[long startIndex, int length]
+        {
+            get
+            {
+                var ret = new List<T>();
+                long endIndex = startIndex + length;
+                int arrayIndex = GetArrayContainingRowIndex(ref startIndex);
+                bool temp = Buffers[arrayIndex][(int)startIndex, length, ret];
+                while (ret.Count < length)
+                {
+                    long nextRowIndex = startIndex + ret.Count + 1;
+                    arrayIndex++;
+                    temp = Buffers[arrayIndex][(int)nextRowIndex, length - ret.Count, ret];
```

Comment:
Is casting `nextRowIndex` to `int`, the right thing here? Same for above where we are casting `startIndex` to an `int`. Why take in a `long` if you are casting it to `int`?

Reply:
nextRowIndex should've been 0. That's a bug. Fixing it.

I'm taking a long because the df can hold long number of elements. Each buffer can hold only intMax number of elements though, so I'm casting once I get to the right buffer

Reply:
```C#
        public IList<T> this[long startIndex, int length]
        {
            get
            {
                var ret = new List<T>(length);
                long endIndex = startIndex + length;
                int arrayIndex = GetArrayContainingRowIndex(ref startIndex);
                bool temp = Buffers[arrayIndex][(int)startIndex, length, ret];
```

Here we take a `long startIndex` and then are casting it `(int)startIndex`. 

I see that `GetArrayContainingRowIndex` is decreasing the number, which honestly is not intuitive.  Maybe make `GetArrayContainingRowIndex` return an `out int newStartIndex` (or a better name). Have `GetArrayContainingRowIndex` ensure that the number is less than `int.MaxValue`, then the 3 calling places don't need to cast here. If it ever fails, I'm not sure what would even happen with the current code.